explorer/interpreter/type_checker.cpp

```diff
@@ -1910,7 +1938,8 @@ auto TypeChecker::SubstituteImpl(const Bindings& bindings,
       Nonnull<const NominalClassType*> new_class_type =
           arena_->New<NominalClassType>(
               &class_type.declaration(),
-              substitute_into_bindings(&class_type.bindings()));
+              substitute_into_bindings(&class_type.bindings()),
+              class_type.base());
```

Comment:
I expect that we need to recursively substitute into this base class type too: call `SubstituteImpl` on the old base and pass the new value into here.

Reply:
Thanks! Done.